src/spaceone/inventory/service/cloud_service_service.py

```diff
@@ -181,30 +176,42 @@ def update(self, params):
         old_cloud_svc_data = dict(cloud_svc_vo.to_dict())
 
         if 'tags' in params:
-            new_tags = self._create_tag(params, tag_type)
+            old_tags = old_cloud_svc_data.get('tags', {})
+            new_tags = self._create_tags(params['tags'], provider)
+            old_tag_keys = old_cloud_svc_data.get('tag_keys', {})
+            new_tag_keys = self._create_tag_keys(params['tags'], provider)
+
+            if new_tags != old_tags:
+                old_tags.update(new_tags)
+                old_tag_keys.update(new_tag_keys)
+                params['tags'] = old_tags
+                params['tag_keys'] = old_tag_keys
+            else:
+                del params['tags']
 
-            new_tags_dict: dict = self._change_tags_to_dict(new_tags)
-            old_tags_dict: dict = self._change_tags_to_dict(old_cloud_svc_data['tags'], tag_type=tag_type)
+        if 'metadata' in params:
+            old_metadata = old_cloud_svc_data.get('metadata', {})
+            new_metadata = self._change_metadata_path(params['metadata'], provider)
```

Comment:
Again, the function name `convert` looks better than `change`.

Reply:
I changed from `_change_metadata_path` to `_convert_metadata` and fixed the logic.
The meaning of function is clearer than before. thank you.